File: cs50/week1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int n, i, j, k, l;

    do
    {
        n = get_int("Height: ");
    }
    //Ensure n is between 1 and 8
    while ((n <= 0) || (n > 8));




    for (i = 0; i < n; i++)
    {
        //Code to produce right-aligned triangle
        for (j = (i+1); j < n; j++)
        {
            printf(" ");
        }

        for (k = 0; k <= i; k++)
        {
            printf("#");
        }

        //Space between triangles
        printf("  ");

        //Code to produce left-aligned triangle
        for (l = 0; l <= i; l++)
        {
            printf("#");
        }


        printf("\n");
    }




}

File: cs50/week6/sentimental-mario-more/mario.py

while True:
    height = int(input("Height: "))
    if height < 9 and height > 0:
        break

for i in range(height):

    # print left middle-aligned triangle
    for j in range(height):

        # start printing # after correct number of spaces
        if j > (height - i):
            print("#", end = "")
        else:
            print(" ", end = "")

        # print space and then right middle-aligned triangle
        if j == height:
            print(" ", "#" * i, end = "")

    # return prompt to new line after each row is completed
    print()


